# Homework Assignment - Lesson 1: Introduction to R

**Student Name:** Vivianna Cowan
**Date:** Feb, 01, 2026 
**Course:** Data Management  

---

## 📚 Instructions

This homework is designed for students with **no prior coding experience**. Work through each section step-by-step, running each cell as you go.

**How to complete this assignment:**
1. Read each section carefully - explanations are provided!
2. **Run each code cell** by clicking the ▶️ play button or pressing `Shift + Enter`
3. For cells that say "YOUR CODE HERE", type your code following the examples
4. Answer the reflection questions in the markdown cells

**💡 Tip:** If you get stuck, refer back to Lecture 1 for examples!

**📋 SUBMISSION**: When you're done, see [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for complete submission instructions.

---

## Part 1: R Basics - Variables and Arithmetic

Before we work with data files, let's practice the fundamentals of R that were covered in Lecture 1. This will help you get comfortable with how R works.

### 1.1 Creating Variables

In R, we store values in **variables** using the `=` sign. Think of a variable like a labeled box that holds a value.

**Example:** `my_age = 25` creates a variable called `my_age` that stores the number 25.

**📝 Task:** Run the cell below to see how variables work:

In [5]:
# Example: Creating variables
# The = sign assigns a value to a variable name
my_number = 24
my_name = "Vivianna Cowan"

# Print the values using the print() function
print(my_number)
print(my_name)

[1] 24
[1] "Vivianna Cowan"


### 1.2 Your Turn: Create Your Own Variables

Now it's your turn! Create the following variables:
- A variable called `favorite_number` with your favorite number
- A variable called `your_name` with your name (remember to put text in quotes!)

**📝 Task:** Replace the `___` with your values and run the cell:

In [6]:
# YOUR CODE HERE - Replace the ___ with your values

favorite_number = 7
your_name = "Vivi"

# Print your variables to see the results
print(favorite_number)
print(your_name)

[1] 7
[1] "Vivi"


### 1.3 Basic Arithmetic

R can work like a calculator! Here are the basic math operations:
- `+` Addition
- `-` Subtraction  
- `*` Multiplication
- `/` Division

**📝 Task:** Run this cell to see arithmetic in action:

In [7]:
# Example: Basic arithmetic
x = 10
y = 5

# Perform calculations and store results
sum_result = x + y
difference = x - y
product = x * y
quotient = x / y

# Print results with labels using paste()
print(paste("x + y =", sum_result))
print(paste("x - y =", difference))
print(paste("x * y =", product))
print(paste("x / y =", quotient))

[1] "x + y = 15"
[1] "x - y = 5"
[1] "x * y = 50"
[1] "x / y = 2"


In [8]:
# YOUR CODE HERE

# Store the number of items sold each day
monday_sales = 15
tuesday_sales = 23

# Calculate total items sold (add both days together)
total_sales = monday_sales + tuesday_sales

# Calculate the difference (how many more items on Tuesday?)
sales_difference = tuesday_sales - monday_sales

# Print the results
print(paste("Total items sold:", total_sales))
print(paste("Tuesday sold more by:", sales_difference))

[1] "Total items sold: 38"
[1] "Tuesday sold more by: 8"


---

## Part 2: Data Structures - Vectors and Data Frames

Now let's learn about how R stores groups of data. This is essential for working with real datasets!

### 2.1 Creating Vectors

A **vector** is a list of values of the same type. We create vectors using the `c()` function (c stands for "combine").

**📝 Task:** Run the cell below to create and view vectors:

In [9]:
# Example: Creating vectors

# A vector of numbers
prices = c(10.99, 25.50, 8.75, 15.00, 32.99)
print("Prices:")
print(prices)

# A vector of text (character strings)
products = c("Apple", "Banana", "Orange", "Grape", "Mango")
print("Products:")
print(products)

[1] "Prices:"
[1] 10.99 25.50  8.75 15.00 32.99
[1] "Products:"
[1] "Apple"  "Banana" "Orange" "Grape"  "Mango" 


### 2.2 Your Turn: Create Vectors

**📝 Task:** Create the following vectors:
- A vector called `test_scores` with 5 test scores (make up numbers between 0-100)
- A vector called `student_names` with 5 student names

In [10]:
# YOUR CODE HERE

# Create a vector of 5 test scores
test_scores = c(98.48, 97.01, 89.99, 67.67, 76.77)

# Create a vector of 5 student names (remember quotes around text!)
student_names = c("Vivi", "Mando", "Layla", "Susy", "Matt")

# Print your vectors
print("Test Scores:")
print(test_scores)
print("Student Names:")
print(student_names)

[1] "Test Scores:"
[1] 98.48 97.01 89.99 67.67 76.77
[1] "Student Names:"
[1] "Vivi"  "Mando" "Layla" "Susy"  "Matt" 


### 2.3 Understanding Data Frames

A **data frame** is like a spreadsheet - it has rows and columns. Each column can contain different types of data. This is the most important data structure for data analysis!

**📝 Task:** Run this cell to create a simple data frame:

In [11]:
# Example: Creating a data frame
employee_data = data.frame(
  Name = c("Alice", "Bob", "Carol"),
  Age = c(28, 35, 42),
  Salary = c(55000, 62000, 78000)
)

# Print the data frame
print("Employee Data:")
print(employee_data)

[1] "Employee Data:"
   Name Age Salary
1 Alice  28  55000
2   Bob  35  62000
3 Carol  42  78000


### 2.4 Your Turn: Create a Data Frame

**📝 Task:** Create a data frame called `my_data` with the following information about 3 of your favorite movies:
- `Title` - the movie name
- `Year` - the year it was released
- `Rating` - your rating from 1-10

In [12]:
# YOUR CODE HERE

# Create a data frame with movie information
my_data = data.frame(
  Title = c("Shrek 2", "The Mummy", "Bugonia"),
  Year = c(2004, 1999, 2025),
  Rating = c(9.99, 8.54, 8.08)
)

# Print your data frame
print("My Favorite Movies:")
print(my_data)

[1] "My Favorite Movies:"
      Title Year Rating
1   Shrek 2 2004   9.99
2 The Mummy 1999   8.54
3   Bugonia 2025   8.08


---

## Part 3: Loading Packages and Importing Data

Now we'll learn how to use R packages and load real data files - this is where the fun begins!

### 3.1 Loading Packages

**Packages** are like apps that add extra features to R. The `tidyverse` package is essential for data analysis.

We load packages using the `library()` function:
```r
library(package_name)
```

You only need to load packages once per session (each time you open the notebook).

**📝 Task:** Write the code to load the `tidyverse` package:

In [13]:
# YOUR CODE HERE
# Load the tidyverse package using the library() function
# Hint: library(package_name) - no quotes needed around the package name

library(tidyverse)

# If successful, you should see a message about packages being loaded
print("Tidyverse loaded successfully!")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.6
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.1     ✔ tibble    3.3.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.2
✔ purrr     1.2.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


[1] "Tidyverse loaded successfully!"


### 3.2 Understanding the Working Directory

The **working directory** is the folder where R looks for files and saves files. Think of it as R's "current location" on your computer.

**Key functions:**
| Function | What It Does | Example |
|----------|--------------|---------|
| `getwd()` | **Get** working directory - shows current location | `getwd()` |
| `setwd()` | **Set** working directory - changes location | `setwd("/path/to/folder")` |

**Why this matters:**
- If R can't find your file, it might be looking in the wrong folder
- Use `getwd()` to see where R is looking
- Use `setwd()` to change to the correct folder

**📝 Task:** Write the code to check your current working directory:

In [14]:
# YOUR CODE HERE
# Use the getwd() function to display your current working directory
# Hint: Just type the function name with empty parentheses

getwd()

[1] "/workspaces/assignment-1-version-3-vivicowan/assignment/Homework"

### 3.3 Setting the Working Directory (If Needed)

If your working directory is not where your files are located, you can change it using `setwd()`.

**Syntax:**
```r
setwd("/path/to/your/folder")
```

**Example paths:**
- Windows: `setwd("C:/Users/YourName/Documents/project")`
- Mac: `setwd("/Users/YourName/Documents/project")`
- This codespace: `setwd("/workspaces/new-clone")`

**💡 Tips:**
- Use forward slashes `/` even on Windows
- The path must be in quotes
- You can also use `setwd("..")` to go up one folder

**📝 Task:** Your working directory should be `/workspaces/new-clone`. If `getwd()` showed something different, write the code to set it correctly:

In [15]:
# YOUR CODE HERE (only if needed)
# If your working directory from 3.2 was NOT /workspaces/new-clone,
# use setwd() to change to the correct directory

setwd("/workspaces/assignment-1-version-3-vivicowan")

# Verify by running getwd() again
getwd()

[1] "/workspaces/assignment-1-version-3-vivicowan"

### 3.4 Import a CSV File

Now that we know where R is looking, let's load some data!

**CSV (Comma-Separated Values)** files are one of the most common data formats. They're simple text files where:
- Each line is a row
- Values are separated by commas

We use the `read_csv()` function from tidyverse:
```r
my_data = read_csv("path/to/file.csv")
```

**📝 Task:** Write the code to import the `company_sales_data.csv` file from the `data/` folder. Store it in a variable called `sales_data`:

In [16]:
# YOUR CODE HERE
# Import the company sales data from the data folder
# Hint: read_csv("folder/filename.csv")
# The file is called "company_sales_data.csv" and it's in the "data" folder

sales_data = read_csv("data/company_sales_data.csv")

# Print a confirmation message
print("Sales data imported successfully!")

Rows: 300 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (3): Sales_Rep_Name, Region, Product_Category
dbl  (4): TransactionID, Revenue, Cost, Units_Sold
date (1): Sale_Date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] "Sales data imported successfully!"


---

## Part 4: Inspecting Your Data

Once you've loaded data, you need to explore it! Here are the essential functions:
- `head()` - shows the first few rows
- `str()` - shows the structure and data types
- `summary()` - shows statistical summaries

### 4.1 View the First Few Rows

The `head()` function shows you the first few rows of your data (default is 6 rows).

**Syntax:**
```r
head(data_frame_name)        # Shows first 6 rows
head(data_frame_name, 10)    # Shows first 10 rows
```

**📝 Task:** Write the code to display the first 10 rows of `sales_data`:

In [17]:
# YOUR CODE HERE
# Use head() to view the first 10 rows of sales_data
# Hint: head(data_name, number_of_rows)

head(sales_data, 10)

TransactionID Sales_Rep_Name Region        Product_Category Revenue 
1   1            Carol Davis    Latin America Services         20750.92
2   2            Carol Davis    Europe        Hardware         32359.98
3   3            Carol Davis    Europe        Services         39268.40
4   4            Bob Smith      Europe        Hardware         28865.09
5   5            Frank Miller   Latin America Software          3932.36
6   6            Carol Davis    Latin America Services         48209.75
7   7            Frank Miller   Europe        Software         33055.35
8   8            Alice Johnson  Europe        Consulting       49856.54
9   9            Bob Smith      Europe        Consulting       13477.13
10 10            Eva Brown      Asia Pacific  Services         21188.96
   Cost     Units_Sold Sale_Date 
1  12253.36 78         2023-04-24
2  24595.20 13         2023-06-09
3  23291.09 34         2023-03-25
4  12428.74 90         2023-04-11
5   1778.18 63         2023-08-26
6  26052.04 26         2023-09-04
7  20432.24 25         2023-06-19
8  37118.63  1         2023-09-01
9   8685.85 20         2023-11-17
10 16757.06 15         2023-08-17

### 4.2 Examine the Data Structure

The `str()` function (short for "structure") tells you:
- How many rows and columns
- The name of each column
- The data type of each column (num = number, chr = text)

**Syntax:**
```r
str(data_frame_name)
```

**📝 Task:** Write the code to view the structure of `sales_data`:

In [18]:
# YOUR CODE HERE
# Use str() to view the structure of sales_data

str(sales_data)

spc_tbl_ [300 × 8] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ TransactionID   : num [1:300] 1 2 3 4 5 6 7 8 9 10 ...
 $ Sales_Rep_Name  : chr [1:300] "Carol Davis" "Carol Davis" "Carol Davis" "Bob Smith" ...
 $ Region          : chr [1:300] "Latin America" "Europe" "Europe" "Europe" ...
 $ Product_Category: chr [1:300] "Services" "Hardware" "Services" "Hardware" ...
 $ Revenue         : num [1:300] 20751 32360 39268 28865 3932 ...
 $ Cost            : num [1:300] 12253 24595 23291 12429 1778 ...
 $ Units_Sold      : num [1:300] 78 13 34 90 63 26 25 1 20 15 ...
 $ Sale_Date       : Date[1:300], format: "2023-04-24" "2023-06-09" ...
 - attr(*, "spec")=
  .. cols(
  ..   TransactionID = col_double(),
  ..   Sales_Rep_Name = col_character(),
  ..   Region = col_character(),
  ..   Product_Category = col_character(),
  ..   Revenue = col_double(),
  ..   Cost = col_double(),
  ..   Units_Sold = col_double(),
  ..   Sale_Date = col_date(format = "")
  .. )
 - attr(*, "problems")=<externalptr

### 4.3 Get Summary Statistics

The `summary()` function gives you statistics for each column:
- For numbers: min, max, mean, median, quartiles
- For text: length, class, mode

**Syntax:**
```r
summary(data_frame_name)
```

**📝 Task:** Write the code to get summary statistics for `sales_data`:

In [19]:
# YOUR CODE HERE
# Use summary() to get summary statistics for sales_data

summary(sales_data)

 TransactionID    Sales_Rep_Name        Region          Product_Category  
 Min.   :  1.00   Length:300         Length:300         Length:300        
 1st Qu.: 75.75   Class :character   Class :character   Class :character  
 Median :150.50   Mode  :character   Mode  :character   Mode  :character  
 Mean   :150.50                                                           
 3rd Qu.:225.25                                                           
 Max.   :300.00                                                           
    Revenue           Cost         Units_Sold       Sale_Date         
 Min.   : 1032   Min.   :  567   Min.   :  1.00   Min.   :2023-01-02  
 1st Qu.:15034   1st Qu.: 7216   1st Qu.: 28.75   1st Qu.:2023-04-07  
 Median :26062   Median :13956   Median : 56.00   Median :2023-07-05  
 Mean   :25906   Mean   :14547   Mean   : 53.90   Mean   :2023-07-07  
 3rd Qu.:37708   3rd Qu.:21714   3rd Qu.: 80.00   3rd Qu.:2023-10-08  
 Max.   :49956   Max.   :37811   Max.   :100.00  

### 4.4 Your Observations

Look at the output from the three cells above and answer these questions:

**How many rows (observations) are in the dataset?**

There are 10 rows of output in the dataset.

**How many columns (variables) are in the dataset?**

There are 8 columns in the dataset.

**What is the highest Revenue value in the dataset?**

The highest revenue value is $49856.54.

**What are the different Regions in the data?**

There are three different regions; europe, latin america, and asia pacific.

---

## Part 5: Working with Your Data

Let's practice using the inspection functions on your own!

### 5.1 Your Turn: Data Inspection Practice

**📝 Task:** Write code to:
1. Show the last 5 rows of sales_data (hint: use `tail()` instead of `head()`)
2. Count how many rows are in the dataset (hint: use `nrow()`)
3. Count how many columns are in the dataset (hint: use `ncol()`)

In [20]:
# YOUR CODE HERE

# Show the last 5 rows
tail(sales_data, 5)

# Count the number of rows
total_rows = nrow(sales_data)
print(paste("Number of rows:", total_rows))

# Count the number of columns
total_cols = ncol(sales_data)
print(paste("Number of columns:", total_cols))

TransactionID Sales_Rep_Name Region        Product_Category Revenue  Cost    
1 296           Alice Johnson  Latin America Services         38026.53 28410.28
2 297           Eva Brown      North America Software         10029.25  4614.59
3 298           Bob Smith      Asia Pacific  Services         37911.51 21905.14
4 299           Alice Johnson  North America Software          2137.30  1156.84
5 300           Carol Davis    Latin America Services         34730.09 21474.69
  Units_Sold Sale_Date 
1 69         2023-08-03
2 61         2023-05-13
3 27         2023-12-22
4 69         2023-12-05
5 39         2023-06-16

[1] "Number of rows: 300"
[1] "Number of columns: 8"


---

## Part 6: Reflection Questions

Answer the following questions based on what you've learned:

### Question 1: Variables

What is the purpose of storing values in variables? Why don't we just use the raw numbers or text directly in our code?

**Your Answer:**

Variables are a tool to store and label values so the code is easier to read and understand. They also make it to reuse values and update them in one place instead of changing the same number or text throughout the entire program.

### Question 2: Data Structures

What is the difference between a vector and a data frame? When would you use each one?

**Your Answer:**

A vector stores a single type of data, like a list of numbers or strings. A data frame stores multiple vectors of different data types in a table-like format, with rows and columns. 

### Question 3: Data Inspection

Why is it important to inspect your data using `head()`, `str()`, and `summary()` before starting analysis? What problems might you discover?

**Your Answer:**

It is important to inspect the data using these functions to understand what the data looks like and how it is structured before analyzing it. It can help by catching problems like incorrect data types, missing values, or unexpected values that could cause errors and lead to incorrect results.

### Question 4: Sales Data Analysis

Based on your inspection of the company sales data:
- What columns would you need to calculate profit for each transaction?
- What analysis could you do with this data in a business context?

**Your Answer:**

The columns needed to calculate profit would be the Revenue and Cost. This data can be used to analyze and identify profitability of a product category or particular products.

---

## ✅ Submission Checklist

Before submitting, make sure you have:

- [ ] Completed Parts 1-2: Created variables, practiced arithmetic, made vectors and data frames
- [ ] Completed Part 3: Loaded the tidyverse package and imported the sales data
- [ ] Completed Part 4: Used head(), str(), and summary() to inspect the data
- [ ] Completed Part 5: Wrote your own code for data inspection
- [ ] Completed Part 6: Answered all four reflection questions
- [ ] **Run all cells** to show your output
- [ ] Saved the notebook

**Great work completing your first R assignment! 🎉**

---

## 🚀 Ready to Submit?

### Easy Submission Steps (No Command Line Required!):

1. **Save this notebook** (Ctrl+S or File → Save)

2. **Use VS Code Source Control**:
   - Click the **Source Control** icon in the left sidebar (tree branch symbol)
   - Click the **"+"** button next to your notebook file
   - Type a message: `Submit homework 1 - [Your Name]`
   - Click **"Commit"** 
   - Click **"Sync Changes"** or **"Push"**

3. **Verify on GitHub**: Go to your repository online and confirm your notebook appears with your completed work

**📖 Need help?** See [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for detailed instructions.

**🎉 Congratulations on completing your first R data analysis assignment!**